tests/contrib/openai/test_logger.py

```diff
@@ -0,0 +1,142 @@
+import os
+import time
+
+import mock
+import pytest
+import vcr
+
+from ddtrace.contrib.openai._logging import V2LogWriter
+from tests.utils import request_token
+
+
+logs_vcr = vcr.VCR(
+    cassette_library_dir=os.path.join(os.path.dirname(__file__), "cassettes/"),
+    record_mode="once",
+    match_on=["path"],
+    filter_headers=[("DD-API-KEY", "XXXXXX")],
+    # Ignore requests to the agent
+    ignore_localhost=True,
+)
+
+
+@pytest.fixture(autouse=True)
+def vcr_logs(request):
+    marks = [m for m in request.node.iter_markers(name="vcr_logs")]
+    assert len(marks) < 2
+    if marks:
+        mark = marks[0]
+        cass = mark.kwargs.get("cassette", request_token(request).replace(" ", "_").replace(os.path.sep, "_"))
+        with logs_vcr.use_cassette("%s.yaml" % cass):
+            yield
+    else:
+        yield
+
+
+@pytest.fixture
+def mock_logs():
+    with mock.patch("ddtrace.contrib.openai._logging.logger") as m:
+        yield m
+
+
+def _test_log():
+    return {
+        "timestamp": time.time() * 1000,
+        "message": "test msg",
+        "hostname": "dev-mbp",
+        "ddsource": "python",
+        "service": "",
+        "status": "info",
+        "ddtags": "",
+    }
+
+
+def test_buffer_limit(mock_logs):
+    logger = V2LogWriter(site="datadoghq.com", api_key="asdf", interval=1000, timeout=1)
+    for i in range(1001):
+        logger.enqueue({})
+    mock_logs.warning.assert_called_with("log buffer full (limit is %d), dropping log", 1000)
+    assert len(logger._buffer) == 1000
+
+
+@pytest.mark.vcr_logs
+def test_send_log(mock_logs):
+    logger = V2LogWriter(site="datadoghq.com", api_key=os.getenv("DD_API_KEY"), interval=1, timeout=1)
+    logger.start()
+    mock_logs.debug.assert_has_calls(
+        [mock.call("started log writer to %r", "https://http-intake.logs.datadoghq.com/api/v2/logs")]
+    )
+    logger.enqueue(_test_log())
+    mock_logs.reset_mock()
```

Comment:
I don't _think_ you need to reset here, since `assert_has_calls` I think checks the call list that all provided calls are there, but not that the list is equal to the call list.

(I'd have to test, but that was my understanding)